samples/bootloader/overlay-minimal-size.conf

```diff
@@ -0,0 +1,21 @@
+#
```

Comment:
IMO it would be good to have a pointer to this file and a brief explanation on how to include it. There should be some examples similar to this already (just grep for 'minimal' in the .rst docs).

Reply:
I've already added such documentation into the `hello_world` sample that will be in a separate PR.

Perhaps it would be simpler to just close this PR and move this addition to that one when it's ready?

Reply:
Got it. I think that information should be available in the bootloader sample documentation as well, since users might not use the hello world sample for information.

Its up to you whether you choose to include it in this PR or another, I have no preference on that.

Reply:
You're right that the information is probably more relevant in this README than the sample. Moving it over here and into this PR works for me :+1: might make the other sample's documentation a bit more concise as well.

Reply:
Added in latest commit. Also opened the PR to be reviewed, since this the majority of the task's content.